website/pages/blog/type-safe-your-api.mdx

```diff
@@ -0,0 +1,291 @@
+  error-prone.
+- Endpoint Interface: REST APIs lack a unified interface for endpoints, making them less intuitive
+  and user-friendly.
+
+## Introducing `feTS`: Type-Safe REST API Development
+
+`feTS`, short for "Fetch" + "TypeScript," is a cutting-edge technology designed to elevate REST API
+development to new heights. By harnessing the capabilities of `TypeScript` and the OpenAPI
+specification, `feTS` ensures that data exchanged between client and server is not only well-defined
+but also strongly typed. This proactive approach catches errors during compilation, resulting in
+more robust and reliable applications.
+
+## A Practical Example: Leveraging `feTS` for Soccer Team Information
+
+To illustrate the power of `feTS`, let's dive into a real-world example. We'll build a demo app that
+interacts with the SportStats V4 API using `feTS`. Our goal is to fetch soccer team information and
+display it in an intuitive user interface.
+
+### Step 1: Obtain a Free Trial Token for Sportstats
+
+To begin, create a free trial Token for SportStats API access. The registration process is quick and
+easy. You can find the registration form [here](https://sportsdata.io/cart/free-trial).
+
+BTW, you can choose any other API you want, but for this example, we will use the SportStats API.
+You can find openApi at `APIS.GURU` [website](https://apis.guru/about/)
+
+### Step 2: Incorporate OpenAPI Definitions into TypeScript
+
+Incorporate the OpenAPI definitions into a TypeScript file and export them using the `as const`
+modifier. This process ensures strong typing and easy integration with `feTS`. A sample code snippet
+is provided below:
+
+```typescript filename="openapi.ts"
+export default { openapi: '3.0.0' /* ... */ } as const
+```
+
+`openapi.ts` GitHub file
+[here](https://github.com/TuvalSimha/feTS-example/blob/main/src/feTS/openapi.ts). You can choose any
+OpenAPI file you want, but for this example, we will use the SportStats OpenAPI file. You can find
+the OpenAPI file [here](https://sportsdata.io/developers/sports-data-open-api-swagger-files).
+
+### Step 3: Install `feTS` and Create a Client
+
+Now, we are ready to install `feTS` and create a new client. To do this, you can visit the docs
+[here](https://the-guild.dev/openapi/`feTS`/client/quick-start#installation).
+
+Next, create a new `TypeScript` file and import the `createClient` function from the `feTS` package.
+Then, use the `createClient` function to instantiate a new client.
+
+```typescript filename="client.ts"
+import { createClient, type NormalizeOAS } from 'feTS'
+import type openapi from './openapi'
+
+export const client = createClient<NormalizeOAS<typeof openapi>>({
+  endpoint: 'https://api.sportsdata.io/v4/soccer/stats'
+})
+```
+
+`client.ts` GitHub file
+[here](https://github.com/TuvalSimha/feTS-example/blob/main/src/feTS/client.ts).
+
+The following code demonstrates how to set up the client with a custom error handling plugin:
```

Comment:
That comes a bit out of nowhere. What's the motivation behind adding the error-handling plugin? That could use some explanation and introduction